### PR TITLE
CI: add scheduled tests

### PR DESCRIPTION
**Pull request by Bastian-Krause**
**Description**
Test results may vary due to new dependency versions. Daily CI runs should help us find issues early.

This becomes especially important once all of labgrid's dependencies get unpinned.

**Checklist**
- [x] PR has been tested

### CODE REVIEW

File: .github/workflows/push-pr-unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+    uses: ./.github/workflows/reusable-unit-tests.yml
+    with:
+      python-version: ${{ matrix.python-version }}
+      branch:
+  push-pr-unit-tests-docker:
+    name: Docker Unit Tests
+    uses: ./.github/workflows/reusable-unit-tests-docker.yml
+    with:
+      branch:
```
Comment (Emantor) [inline .github/workflows/push-pr-unit-tests.yml:20]: This looks weird, is this required?

Comment (Bastian-Krause) [reply .github/workflows/push-pr-unit-tests.yml:20]: Passing an empty branch makes `actions/checkout@v3` use the default one. That's what we want here. If we mark the input as `required: false`, we can probably drop this. I'd have to look what is passed in that case.

Comment (Bastian-Krause) [reply .github/workflows/push-pr-unit-tests.yml]: Fixed.

**Comment by Bastian-Krause** [comment]
Force-pushed:
- made branch input optional, default is an empty string, see [docs](https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_callinputs)
- dropped empty branch inputs

**Review by Emantor — APPROVED** [review]

File: .github/workflows/push-pr-unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: unit tests
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  push-pr-unit-tests:
```
Comment (Emantor) [inline .github/workflows/push-pr-unit-tests.yml:6]: Isn't the trick here to name this `build` so it still fits the repo settings? This way we shouldn't need to reconfigure the repo and rebase old PRs?

File: .github/workflows/push-pr-unit-tests.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    name: unit tests
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+    uses: ./.github/workflows/reusable-unit-tests.yml
+    with:
+      python-version: ${{ matrix.python-version }}
+  push-pr-unit-tests-docker:
```
Comment (Emantor) [inline .github/workflows/push-pr-unit-tests.yml:15]: Same here, shouldn't this be named `docker`?

**Review by Emantor — CHANGES_REQUESTED** [review]

**Comment by Emantor** [comment]
Ah, I guess this may be more than simple string matching, but I'll test it nevertheless.

**Comment by Bastian-Krause** [comment]
The checks now have a 3 component identifier, previously it consisted of 2 components. I don't know if renaming makes a difference. You can try that, though. I don't have the access rights to see the configuration of the required checks anyway.

**Comment by Emantor** [comment]
It doesn't matter, the extra component removes the match.

**Comment by jluebbe** [comment]
LGTM, but we should check if we can keep the existing check associations.

**Comment by jluebbe** [comment]
Due to the different workflow call chain, it doesn't seem possible to keep the check names the same.

**Review by jluebbe — APPROVED** [review]

**Pull request merged**